--- a/scipy/_lib/_ccallback_c.pyx
+++ b/scipy/_lib/_ccallback_c.pyx
@@ -201,19 +201,22 @@ cdef double sine(double x, void *user_data) except * nogil:
     return sin(x)
 
 
-# Ctypes declarations of the callables above
-import ctypes
-
-plus1_t = ctypes.CFUNCTYPE(ctypes.c_double, ctypes.c_double, ctypes.POINTER(ctypes.c_int), ctypes.c_void_p)
-plus1_ctypes = ctypes.cast(<size_t>&plus1_cython, plus1_t)
-
-plus1b_t = ctypes.CFUNCTYPE(ctypes.c_double, ctypes.c_double, ctypes.c_double,
-                            ctypes.POINTER(ctypes.c_int), ctypes.c_void_p)
-plus1b_ctypes = ctypes.cast(<size_t>&plus1b_cython, plus1b_t)
-
-plus1bc_t = ctypes.CFUNCTYPE(ctypes.c_double, ctypes.c_double, ctypes.c_double, ctypes.c_double,
-                            ctypes.POINTER(ctypes.c_int), ctypes.c_void_p)
-plus1bc_ctypes = ctypes.cast(<size_t>&plus1bc_cython, plus1bc_t)
-
-sine_t = ctypes.CFUNCTYPE(ctypes.c_double, ctypes.c_double, ctypes.c_void_p)
-sine_ctypes = ctypes.cast(<size_t>&sine, sine_t)
+# Ctypes declarations of the callables above (test fixtures only)
+try:
+    import ctypes
+
+    plus1_t = ctypes.CFUNCTYPE(ctypes.c_double, ctypes.c_double, ctypes.POINTER(ctypes.c_int), ctypes.c_void_p)
+    plus1_ctypes = ctypes.cast(<size_t>&plus1_cython, plus1_t)
+
+    plus1b_t = ctypes.CFUNCTYPE(ctypes.c_double, ctypes.c_double, ctypes.c_double,
+                                ctypes.POINTER(ctypes.c_int), ctypes.c_void_p)
+    plus1b_ctypes = ctypes.cast(<size_t>&plus1b_cython, plus1b_t)
+
+    plus1bc_t = ctypes.CFUNCTYPE(ctypes.c_double, ctypes.c_double, ctypes.c_double, ctypes.c_double,
+                                ctypes.POINTER(ctypes.c_int), ctypes.c_void_p)
+    plus1bc_ctypes = ctypes.cast(<size_t>&plus1bc_cython, plus1bc_t)
+
+    sine_t = ctypes.CFUNCTYPE(ctypes.c_double, ctypes.c_double, ctypes.c_void_p)
+    sine_ctypes = ctypes.cast(<size_t>&sine, sine_t)
+except ImportError:
+    pass
